proxy-node-chart/charts/metadata-controller/templates/statefulset.yaml

```diff
@@ -0,0 +1,87 @@
+apiVersion: apps/v1
+kind: StatefulSet
+metadata:
+  name: {{ include "metadata-controller.fullname" . }}
+  labels:
+    app.kubernetes.io/name: {{ include "metadata-controller.name" . }}
+    helm.sh/chart: {{ include "metadata-controller.chart" . }}
+    app.kubernetes.io/instance: {{ .Release.Name }}
+    app.kubernetes.io/managed-by: {{ .Release.Service }}
+  {{- with .Values.annotations }}
+  annotations:
+    {{- toYaml . | nindent 4 }}
+  {{- end }}
+spec:
+  replicas: {{ .Values.replicaCount }}
+  serviceName: {{ include "metadata-controller.fullname" . }}
+  selector:
+    matchLabels:
+      app.kubernetes.io/name: {{ include "metadata-controller.name" . }}
+      app.kubernetes.io/instance: {{ .Release.Name }}
+  template:
+    metadata:
+      labels:
+        app.kubernetes.io/name: {{ include "metadata-controller.name" . }}
+        app.kubernetes.io/instance: {{ .Release.Name }}
+    spec:
+      containers:
+      - name: {{ .Chart.Name }}-cloudhsm-client
```

Comment:
As discussed the cloudhsm client should be a separate deployment/pod.

Reply:
we decided to place cloudhsm client in sibling container afterall